library/std/src/sys/unix/os.rs

```diff
@@ -566,10 +583,16 @@ pub fn unsetenv(n: &OsStr) -> io::Result<()> {
     }
 }
 
+#[cfg(not(target_os = "espidf"))]
 pub fn page_size() -> usize {
     unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize }
 }
 
+#[cfg(target_os = "espidf")]
+pub fn page_size() -> usize {
+    32
```

Comment:
What is the significance of this number? AFAIK RISC-V uses 4K pages. A comment here would be useful.

Reply:
Using the RISC-V standard of 4K pages might not make sense when add support for the Xtensa based Espressif chips (once support is upstreamed in LLVM of course). 

I also believe setting a smaller page size reduces the stack usage of threads where the stack size is not a multiple of the page size, according to the `Thread::new` impl: https://github.com/rust-lang/rust/blob/6fe0886723c9e08b800c9951f1c6f6a57b2bf22c/library/std/src/sys/unix/thread.rs#L63-L66

So I think we want to keep this number low.

Reply:
Since none of the chips supported by esp-idf so far has an MMU, we can indeed keep this number low. Perhaps 4 bytes would work?

However it is odd that this chip-specific property ends up depending on `#[cfg(target_os = "espidf")]`.

Long term, my preference would be implementing `sysconf`  and `_SC_PAGESIZE` in ESP-IDF and deciding on the right value there (possibly depending on the chip target).

Reply:
If this is just about the stack alignment then maybe it would be better to create a `stack_alignment` function in the thread module which can return `page_size` by default but a different value for `espidf`.

Reporting a page size of 32 is just confusing, even if this function is internal to `std`.

Reply:
Please see [this summary of my latest changes](https://github.com/rust-lang/rust/pull/87666#issuecomment-893460598). No more `page_size` on ESP-IDF.